src/shacl2code/lang/golang.py

```diff
@@ -2,153 +2,300 @@
 
-def interface_setter(prop):
-    return "Set" + type_name(prop.varname) + '(' + setter_prop_type(prop) + ') error'
+    def constant_var_name(self,named):
+        if self.uppercase_constants:
+            return upper_first(named.varname)
+        return named.varname
 
+    def getter_name(self,prop):
+        return self.getter_prefix + upper_first(self.struct_prop_name(prop))
 
-def struct_getter(cls, prop):
-    return 'func (o *' + struct_name(cls) + ') ' + type_name(prop.varname) + '() ' + prop_type(prop) + '{\n' \
-        + '    return o.' + struct_prop_name(prop) + '\n' \
-        + '}'
+    def setter_name(self,prop):
+        return self.setter_prefix + upper_first(self.struct_prop_name(prop))
 
+    def concrete_name(self,cls):
+        if self.export_structs:
+            return self.struct_name(cls)
+        if self.use_embedding:
+            return self.embedded_prefix + upper_first(self.struct_name(cls))
+        if cls.is_abstract:
+            return self.struct_name(cls)
+        return upper_first(self.struct_name(cls))
 
-def struct_setter(cls, prop):
-    return 'func (o *' + struct_name(cls) + ') Set' + type_name(prop.varname) + '(v ' + setter_prop_type(prop) + ') error{\n' \
-        + '    o.' + struct_prop_name(prop) + '  = v\n' \
-        + '    return nil\n' \
-        + '}'
+    def trim_package(self,code):
+        return code.replace("package v3_0", "")
 
 
-def comment(indent_with, identifier, text):
-    if text.lower().startswith(identifier.lower()):
-        text = identifier + " " + text[len(identifier):]
+# common utility functions
 
-    return indent(indent_with, text)
 
-def parent_has_prop(classes, cls, prop):
-    for parent_id in cls.parent_ids:
-        parent = classes.get(parent_id)
-        for p in parent.properties:
-            if p.varname == prop.varname:
-                return True
-        if parent_has_prop(classes, parent, prop):
-            return True
+def upper_first(str):
+    return str[0].upper() + str[1:]
 
-    return False
 
+def lower_first(str):
+    return str[0].lower() + str[1:]
 
-def include_prop(classes, cls, prop):
-    return not parent_has_prop(classes, cls, prop)
 
+def indent(indent_with, str):
+    parts = re.split("\n", str)
+    return indent_with + ("\n"+indent_with).join(parts)
 
 
+def dedent(str, amount):
+    prefix = ' ' * amount
+    parts = re.split("\n", str)
+    for i in range(len(parts)):
+        if parts[i].startswith(prefix):
+            parts[i] = parts[i][len(prefix):]
+    return '\n'.join(parts)
 
-def interface_name(cls):
-    return type_name(cls.clsname)
 
+def type_name(name):
+    if isinstance(name, list):
+        name = "".join(name)
+    parts = re.split(r'[^a-zA-Z0-9]', name)
+    part = parts[len(parts)-1]
+    return upper_first(part)
 
 
+def gofmt(code):
+    try:
+        proc = subprocess.Popen(["gofmt"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE, text=True)
```

Comment:
You could just use `subprocess.run(["gofmt"], capture_output=true)`. It's thread safe.